pkg/controller/common/reconciler.go

```diff
@@ -198,11 +203,33 @@ func (r *BaseIngressReconciler) reconcileRoute(ctx context.Context, ci networkin
 }
 
 func (r *BaseIngressReconciler) reconcileDeletion(ctx context.Context, ci networkingv1alpha1.IngressAccessor) error {
-	// TODO: something with a finalizer?  We're using owner refs for
```

Comment:
Do we need to check first if the current finalizer is the first? I've seen other code in Knative do that and I guess it's to guarantee that no two finalizers step on each others feet.

Reply:
For reference: https://github.com/knative/serving/blob/0d22ddd171a017b673baa34dbe90d5d6d219c587/pkg/reconciler/ingress/ingress.go#L427-L450